src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -294,14 +295,20 @@ public <R> Variable<R> getConvertedExpression(Class<R>... to) {
 	 * Gets the value of this variable as stored in the variables map.
 	 */
 	@Nullable
-	public Object getRaw(Event e) {
-		String n = name.toString(e);
-		if (n.endsWith(Variable.SEPARATOR + "*") != list) // prevents e.g. {%expr%} where "%expr%" ends with "::*" from returning a Map
+	public Object getRaw(Event event) {
+		String name = this.name.toString(event);
+		if (name.endsWith(Variable.SEPARATOR + "*") != list) // prevents e.g. {%expr%} where "%expr%" ends with "::*" from returning a Map
 			return null;
-		Object val = !list ? convertIfOldPlayer(n, e, Variables.getVariable(n, e, local)) : Variables.getVariable(n, e, local);
-		if (val == null)
-			return Variables.getVariable((local ? LOCAL_VARIABLE_TOKEN : "") + name.getDefaultVariableName(), e, false);
-		return val;
+		Object value = !list ? convertIfOldPlayer(name, event, Variables.getVariable(name, event, local)) : Variables.getVariable(name, event, local);
+		// Check for default variables.
+		if (value == null) {
+			for (String typeHint : this.name.getDefaultVariableNames(event)) {
+				value = Variables.getVariable((local ? LOCAL_VARIABLE_TOKEN : "") + typeHint, event, local);
```

Comment:
This will make variable getting much slower, as an expression like `{something::%player%}` can result in 9 getVariable calls (if the var isn't set and there isn't a default variable for it):
![image](https://user-images.githubusercontent.com/29547183/175820868-f97d0409-d0b1-46be-8558-33db28c6e64a.png)
and something like `{relation::%player%::%{_p}%}` in 65 getVariable calls, it goes up exponentially

This system should be more efficient, I don't think such a performance cost is worth it.

Reply:
Yes I did take this into consideration, but it's already doing this now. I had a thought to somehow have a scriptinfo reference and check if the script had default variables, but that's alot of implementation. Do you have an alternative solution?